src/allmydata/test/common.py

```diff
@@ -1147,6 +1153,28 @@ def _corrupt_uri_extension(data, debug=False):
     return corrupt_field(data, 0x0c+uriextoffset, uriextlen)
 
 
+
+@attr.s
+@implementer(IAddressFamily)
+class FakeProvider(object):
```

Comment:
maybe 

```suggestion
class FixedListenerAndEndpoitProvider(object):
```

it's a bit long, but it described what it does :) ... Fake is generic and you can fake int in many ways.